### PR TITLE
Homework7

### PR DESCRIPTION
**Pull request by featherko**

### CODE REVIEW

File: homework/hw7/task1_recdict.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    exam = {a: b for a, b in zip(range(3), range(3))}
+
+    print(find_occurrences_rec(exam, exam))  # noqa: T001
+    z = {**exam, "a": exam}
+    z = {**exam, "a": z}
+    print(z)  # noqa: T001
+    print(exam)  # noqa: T001
+    print((find_occurrences_rec(z, exam)))  # noqa: T001
+
+"""for key, value in tree.items():
```
Comment (Aleksei-Samarin) [inline homework/hw7/task1_recdict.py]: It's better to leave comments here on GitHub in the pull request with concrete questions

File: homework/hw7/task1_recdict.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+if __name__ == "__main__":
+    print(find_occurrences_rec(example_tree, "RED"))  # noqa: T001
+
+    exam = {a: b for a, b in zip(range(3), range(3))}
+
+    print(find_occurrences_rec(exam, exam))  # noqa: T001
+    z = {**exam, "a": exam}
+    z = {**exam, "a": z}
+    print(z)  # noqa: T001
```
Comment (Aleksei-Samarin) [inline homework/hw7/task1_recdict.py]: We can transform all these # noqa: T001 to one rule for exact file to get more readability

File: homework/hw7/task2_backspace.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+def backspace_compare(first: str, second: str) -> bool:
+    """Compare if two given strings are equal."""
+    return c(first) == c(second)
+
+
+"""some proof of concept"""
+
+
+def random_fun(string: str, string2: str) -> bool:  # noqa: CCR001
```
Comment (Aleksei-Samarin) [inline homework/hw7/task2_backspace.py:39]: Can we simplify the solution?

File: test/hw7/test_task1_recdict.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from typing import NoReturn, Optional
+
+import pytest
+
+from homework.hw7.task1_recdict import find_occurrences
+
+
+example_tree = {
```
Comment (Aleksei-Samarin) [inline test/hw7/test_task1_recdict.py]: Use resource files for JSON like objects. It's a bad practice to use them in .py files. Also in tests for that purpose we can use fixtures

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

File: homework/hw7/task2_backspace.py [hunk 5]
```diff
@@ -27,50 +28,42 @@ def calc_backspace(text: str) -> str:
...
     """Compare if two given strings are equal."""
-    return c(first) == c(second)
+    return calc_backspace(first) == calc_backspace(second)
 
 
 """some proof of concept"""
 
 
 def random_fun(string: str, string2: str) -> bool:  # noqa: CCR001
     """Compare if two given strings are equal."""
```
Comment (featherko) [inline homework/hw7/task2_backspace.py:40]: Not sure if there anything else to simplify.

File: test/hw7/test_task1_recdict.py [hunk 6]
```diff
@@ -1,34 +1,25 @@
...
-            "key2": "RED",
-            "key3": ["a", "lot", "of", "values", {"nested_key": "RED"}],
-        },
-    },
-    "fourth": "RED",
-}
+def example_tree_build():
+    with open(os.path.join(os.getcwd(), "test", "hw7", "tree.json")) as f:
+        return json.load(f)
+
```
Comment (featherko) [inline test/hw7/test_task1_recdict.py:13]: Is that right way to implement it? and doesn't @pytest.mark.parametrize already making fixtures from what it takes inside it?

File: homework/hw7/task2_backspace.py (hunk 5 above)
Comment (Aleksei-Samarin) [reply homework/hw7/task2_backspace.py:40]: It's ok to use a separated function to split complex solution

File: test/hw7/test_task1_recdict.py (hunk 6 above)
Comment (Aleksei-Samarin) [reply test/hw7/test_task1_recdict.py:13]: With fixtures we can get some extra possibilities such as session scopes to preparare object only once

**Review by Aleksei-Samarin — APPROVED** [review]

**Pull request merged**